File: lc206/lc206.go
package main

import . "LeetCode/ListNode"

func reverseList(head *ListNode) *ListNode {
	cur := head
	var prev *ListNode
	for cur != nil {
		// temp := cur.Next
		// cur.Next = prev
		// prev = cur
		// cur = temp
		cur.Next, prev, cur = prev, cur, cur.Next
	}
	return prev
}

func main() {
	head := MakeList([]int{1, 2, 3, 4, 5, 6, 7})
	PrintList(head)
	rhead := reverseList(head)
	PrintList(rhead)
}

File: lc206/lc206.java
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode cur = head;
        ListNode prev = null;
        while (cur != null) {
            ListNode next = cur.next;
            cur.next = prev;
            prev = cur;
            cur = next;
        }
        return prev;
    }
}

private class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }
}
